Content.Server/GameTicking/GameTicker.Player.cs

```diff
@@ -185,6 +189,33 @@ async void AddPlayerToDb(Guid id)
             }
         }
 
+        private void AdminAlertIfSharedConnection(ICommonSession newSession, IPAddress addr)
+        {
+            var playerThreshold = _cfg.GetCVar(CCVars.AdminAlertMinPlayersSharingConnection);
+            if (playerThreshold < 0)
+                return;
+
+            var sharedConnections = _playerManager.Sessions.Where(session =>
+                    session.Channel.IsHandshakeComplete
+                    && session.Channel.RemoteEndPoint.Address.Equals(addr))
+                .ToList();
+
+            var shareConnectionCount = sharedConnections.Count();
+            if (shareConnectionCount < playerThreshold)
+                return;
+
+            var username = newSession.Name;
+            var otherUsernames = string.Join(", ",
+                sharedConnections.Where(session =>
+                        session.UserId != newSession.UserId)
+                    .Select(session => _playerManager.GetSessionById(session.UserId).Name));
```

Comment:
Uh, why call `_playerManager.GetSessionById(session.UserId)`? You already have the session.